tool/src/dart/mod.rs

```diff
@@ -401,22 +403,24 @@ impl<'a, 'cx> TyGenContext<'a, 'cx> {
                         if let MaybeStatic::NonStatic(lt) = lt {
                             if method_lifetime.all_longer_lifetimes.contains(&lt) {
                                 let edge = if let hir::Type::Slice(..) = ty {
-                                    // Slices make a temporary view type that needs to be attached
-                                    // XXXManishearth: this is the wrong variable. We need to grab on to the arena
-                                    // and also ensure it's not destroyed.
-                                    format!("{param_name}View")
+                                    // Slices make an arena with a finalizer that needs to be attached
+                                    format!("{param_name}Arena")
```

Comment:
If we are doing one arena per parameter we might as well do the same as in JS where we attach finalizers that free using `diplomat_free`

Reply:
What's the difference? Both allocate in the native heap, yes?

Also it's one arena per parameter that gets borrowed in the output. It's going to be rare.

I don't think it's a good idea to have two separate paths for slice construction based on arenas or native rust allocation: the way this is designed all slices need for conversion is an arena, it can be a temporary one or a finalized one.